components/Ribbon/src/Ribbon.cs

```diff
@@ -0,0 +1,201 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Specialized;
+
+namespace CommunityToolkit.WinUI.Controls;
+
+/// <summary>
+/// An Office like Ribbon control which displays groups of commands. If there is not enough space to display all the groups,
+/// some of them can be collapsed based on a priority order.
+/// </summary>
+[ContentProperty(Name = nameof(Items))]
+[TemplatePart(Name = PanelTemplatePart, Type = typeof(Panel))]
+[TemplatePart(Name = ScrollViewerTemplatePart, Type = typeof(ScrollViewer))]
+[TemplatePart(Name = ScrollDecrementButtonTempatePart, Type = typeof(ButtonBase))]
+[TemplatePart(Name = ScrollIncrementButtonTempatePart, Type = typeof(ButtonBase))]
+[TemplateVisualState(GroupName = ScrollButtonGroupNameTemplatePart, Name = NoButtonsStateTemplatePart)]
+[TemplateVisualState(GroupName = ScrollButtonGroupNameTemplatePart, Name = DecrementButtonStateTemplatePart)]
+[TemplateVisualState(GroupName = ScrollButtonGroupNameTemplatePart, Name = IncrementButtonStateTemplatePart)]
+[TemplateVisualState(GroupName = ScrollButtonGroupNameTemplatePart, Name = BothButtonsStateTemplatePart)]
+public sealed partial class Ribbon : Control
+{
+    private const string PanelTemplatePart = "Panel";
+    private const string ScrollViewerTemplatePart = "ScrollViewer";
+    private const string ScrollDecrementButtonTempatePart = "ScrollDecrementButton";
+    private const string ScrollIncrementButtonTempatePart = "ScrollIncrementButton";
+    private const string ScrollButtonGroupNameTemplatePart = "ScrollButtonGroup";
+    private const string NoButtonsStateTemplatePart = "NoButtons";
+    private const string DecrementButtonStateTemplatePart = "DecrementButton";
+    private const string IncrementButtonStateTemplatePart = "IncrementButton";
+    private const string BothButtonsStateTemplatePart = "BothButtons";
+
+    private Panel? _panel;
+    private ScrollViewer? _scrollViewer;
+    private ButtonBase? _decrementButton;
+    private ButtonBase? _incrementButton;
+    private readonly ObservableCollection<UIElement> _items;
+
+    /// <summary>
+    /// The DP to store the <see cref="ScrollStep"/> property value.
+    /// </summary>
+    public static readonly DependencyProperty ScrollStepProperty = DependencyProperty.Register(
+        nameof(ScrollStep),
+        typeof(double),
+        typeof(Ribbon),
+        new PropertyMetadata(20.0));
+
+    /// <summary>
+    /// The amount to add or remove from the current scroll position.
+    /// </summary>
+    public double ScrollStep
+    {
+        get => (double)GetValue(ScrollStepProperty);
+        set => SetValue(ScrollStepProperty, value);
+    }
+
+    public Ribbon()
+    {
+        DefaultStyleKey = typeof(Ribbon);
+
+        _items = [];
+        _items.CollectionChanged += OnItemsCollectionChanged;
+    }
+
+    public ICollection<UIElement> Items => _items;

```

Comment:
Wonder if this should be `IList`, that may not matter with the setup you have for this here, but I know we've had other XAML related issues for other collection type controls we've had in the past as some internal XAML pieces expect IList over other collection types.